php-frontend/src/main/java/org/sonar/php/parser/NewPHPGrammar.java

```diff
@@ -0,0 +1,1479 @@
+      f.functionDeclaration(
+        b.token(PHPKeyword.FUNCTION),
+        b.optional(b.token(PHPPunctuator.AMPERSAND)),
+        IDENTIFIER(),
+        PARAMETER_LIST(),
+        BLOCK()));
+  }
+
+  public ParameterListTree PARAMETER_LIST() {
+    return b.<ParameterListTree>nonterminal(PHPLexicalGrammar.PARAMETER_LIST).is(
+      f.parameterList(
+        b.token(PHPPunctuator.LPARENTHESIS),
+        b.optional(
+          f.newTuple94(
+            PARAMETER(),
+            b.zeroOrMore(
+              f.newTuple93(
+                b.token(PHPPunctuator.COMMA),
+                PARAMETER())))),
+        b.token(PHPPunctuator.RPARENTHESIS)));
+  }
+
+  public ParameterTree PARAMETER() {
+    return b.<ParameterTree>nonterminal(PHPLexicalGrammar.PARAMETER).is(
+      f.parameter(
+        b.optional(
+          b.firstOf(
+            b.token(PHPKeyword.ARRAY),
+            b.token(PHPKeyword.CALLABLE),
+            NAMESPACE_NAME())),
+        b.optional(b.token(PHPPunctuator.AMPERSAND)),
+        b.optional(b.token(PHPPunctuator.ELIPSIS)),
+        b.token(PHPLexicalGrammar.REGULAR_VAR_IDENTIFIER),
+        b.optional(
+          f.newTuple92(
+            b.token(PHPPunctuator.EQU),
+            STATIC_SCALAR()))));
+  }
+
+  public SeparatedListImpl<NamespaceNameTree> INTERFACE_LIST() {
+    return b.<SeparatedListImpl<NamespaceNameTree>>nonterminal(PHPLexicalGrammar.INTERFACE_LIST).is(
+      f.interfaceList(
+        NAMESPACE_NAME(),
+        b.zeroOrMore(f.newTuple98(b.token(COMMA), NAMESPACE_NAME()))));
+  }
+
+  public UseTraitDeclarationTree USE_TRAIT_DECLARATION() {
+    return b.<UseTraitDeclarationTree>nonterminal(Kind.USE_TRAIT_DECLARATION).is(
+      b.firstOf(
+        f.useTraitDeclaration(
+          b.token(PHPKeyword.USE),
+          INTERFACE_LIST(),
+          EOS()),
+        f.useTraitDeclaration(
+          b.token(PHPKeyword.USE),
+          INTERFACE_LIST(),
+          b.token(LCURLYBRACE),
+          b.zeroOrMore(
+            b.firstOf(
+              TRAIT_PRECEDENCE(),
+              TRAIT_ALIAS())),
+          b.token(RCURLYBRACE))));
+  }
+
+  public TraitPrecedenceTree TRAIT_PRECEDENCE() {
+    return b.<TraitPrecedenceTree>nonterminal(PHPLexicalGrammar.TRAIT_PRECEDENCE).is(
+      f.traitPrecedence(
+        TRAIT_METHOD_REFERENCE_FULLY_QUALIFIED(),
+        b.token(PHPKeyword.INSTEADOF),
+        INTERFACE_LIST(),
+        EOS()));
+  }
+
+  public TraitAliasTree TRAIT_ALIAS() {
+    return b.<TraitAliasTree>nonterminal(PHPLexicalGrammar.TRAIT_ALIAS).is(
+      b.firstOf(
+        f.traitAlias(
+          TRAIT_METHOD_REFERENCE(),
+          b.token(PHPKeyword.AS),
+          b.optional(MEMBER_MODIFIER()),
+          IDENTIFIER(),
+          EOS()),
+        f.traitAlias(
+          TRAIT_METHOD_REFERENCE(),
+          b.token(PHPKeyword.AS),
+          MEMBER_MODIFIER(),
+          EOS())));
+  }
+
+  public TraitMethodReferenceTree TRAIT_METHOD_REFERENCE() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00100)